test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,128 @@ def test_reservations_with_expired_subjobs(self):
         # Check status of the parent-job using qstat -fx once reservation ends
         self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
                            extend='x', id=jid)
+
+    def common_config(self):
+        """
+        This function contains common steps for test
+        "test_ASAP_resv_with_multinode_job_array" and
+        "test_standing_resv_with_multinode_job_array"
+        """
+        vn_attrs = {ATTR_rescavail + '.ncpus': 4}
+        self.server.create_vnodes("vnode1", vn_attrs, 1,
+                                  self.mom, fname="vnodedef1")
+        self.server.create_vnodes("vnode2", vn_attrs, 2,
+                                  self.mom, additive=True, fname="vnodedef2")
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+
+    def test_ASAP_resv_with_multivnode_job_array(self):
+        """
+        Test multivnode job array with ASAP reservation. Also
```

Comment:
I personally think this test is not needed. Lovely's test would cover the scenario where array jobs are submitted to reservations (advance or standing), I'm not sure what we gain from submitting multi-vnode (or multi-node) jobs in this test suite.

Reply:
@arungrover This test was added as part of regression testing.